src/WattleScript.Interpreter/Tree/Expressions/BinaryOperatorExpression.cs

```diff
@@ -172,64 +179,31 @@ private static Expression CreateSubTree(LinkedList list, ScriptLoadingContext lc
 			return nodes.Expr;
 		}
 
-		private static Node PrioritizeLeftAssociative(Node nodes, ScriptLoadingContext lcontext, Operator operatorsToFind)
-		{
-			for (Node N = nodes; N != null; N = N.Next)
-			{
-				Operator o = N.Op;
-
-				if ((o & operatorsToFind) != 0)
-				{
-					N.Op = Operator.NotAnOperator;
-					N.Expr = new BinaryOperatorExpression(N.Prev.Expr, N.Next.Expr, o, lcontext);
-					N.Prev = N.Prev.Prev;
-					N.Next = N.Next.Next;
-
-					if (N.Next != null)
-						N.Next.Prev = N;
-
-					if (N.Prev != null)
-						N.Prev.Next = N;
-					else
-						nodes = N;
-				}
-			}
-
-			return nodes;
-		}
-
-		private static Node PrioritizeRightAssociative(Node nodes, ScriptLoadingContext lcontext, Operator operatorsToFind)
+		private static Node PrioritizeAssociative(Associativity side, Node node, ScriptLoadingContext lcontext, Operator operatorsToFind)
 		{
-			Node last;
-			for (last = nodes; last.Next != null; last = last.Next)
+			for (Node n = side == Associativity.Left ? node : node.GetLast(); n != null; n = side == Associativity.Left ? n.Next : n.Prev)
```

Comment:
This line is a bit of a mess. I would suggest changing that into a couple of statements + a while loop to make it more clear what is going on